src/textual/app.py

```diff
@@ -716,27 +716,22 @@ def update_styles(self, node: DOMNode | None = None) -> None:
         self._require_stylesheet_update.add(self.screen if node is None else node)
         self.check_idle()
 
-    def mount(self, *anon_widgets: Widget, **widgets: Widget) -> AwaitMount:
-        """Mount widgets. Widgets specified as positional args, or keywords args. If supplied
-        as keyword args they will be assigned an id of the key.
+    def mount(self, *widgets: Widget) -> AwaitMount:
+        """Mount widgets.
 
         Returns:
             AwaitMount: An awaitable object that waits for widgets to be mounted.
 
         """
-        mounted_widgets = self._register(self.screen, *anon_widgets, **widgets)
-        return AwaitMount(mounted_widgets)
+        return AwaitMount(self._register(self.screen, *widgets))
 
     def mount_all(self, widgets: Iterable[Widget]) -> AwaitMount:
         """Mount widgets from an iterable.
 
         Args:
             widgets (Iterable[Widget]): An iterable of widgets.
         """
-        mounted_widgets = list(widgets)
-        for widget in mounted_widgets:
-            self._register(self.screen, widget)
-        return AwaitMount(mounted_widgets)
+        return self.mount(*list(widgets))
```

Comment:
The `list` is not needed if you unpack the widgets anyway.
```suggestion
        return self.mount(*widgets)
```

Reply:
Thanks for pointing that out @UriyaHarpeness; keep in mind this PR is a draft, currently an ongoing work in progress.